app/components/primer/state_component.rb

```diff
@@ -4,12 +4,19 @@ module Primer
   # Component for rendering the status of an item.
   class StateComponent < Primer::Component
     COLOR_DEFAULT = :default
-    COLOR_MAPPINGS = {
+    NEW_COLOR_MAPPINGS = {
+      open: "State--green",
+      closed: "State--red",
+      merged: "State--purple"
+    }.freeze
+
+    DEPRECATED_COLOR_MAPPINGS = {
       COLOR_DEFAULT => "",
       :green => "State--green",
       :red => "State--red",
       :purple => "State--purple"
     }.freeze
+    COLOR_MAPPINGS = NEW_COLOR_MAPPINGS.merge(DEPRECATED_COLOR_MAPPINGS)
     COLOR_OPTIONS = COLOR_MAPPINGS.keys
 
     SIZE_DEFAULT = :default
```

Comment:
Can we also change the examples to use functional colors?

Reply:
Done in ce30e41e45768ee96550ddf8766290cbcc5ad2ff!